ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -806,6 +811,16 @@ public static boolean isChunkUpload(Map<String, Object> args) throws RestService
     return getBooleanHeader(args, Headers.CHUNK_UPLOAD, false);
   }
 
+  /**
+   * Determine if {@link Headers#PUT_UPDATE} is set in the request args.
+   * @param args The request arguments.
+   * @return {@code true} if {@link Headers#PUT_UPDATE} is set.
+   * @throws RestServiceException if exception occurs during parsing the arg.
+   */
+  public static boolean isUpsert(Map<String, Object> args) throws RestServiceException {
```

Comment:
BTW, RESTful PUT is supposed to be insert-if-missing and update-if-exists. That means PUT itself means upsert.  Why do we need an extra header to indicate that?

Reply:
Yeah, updated. By default the update for named-blob will be upsert. User can specify a header 'x-ambry-named-not-upsert' to disable this default upsert (see the latest commit).